contractcourt/chain_arbitrator.go

```diff
@@ -397,6 +397,32 @@ func (c *ChainArbitrator) Start() error {
 		}
 
 		c.activeChannels[chanPoint] = channelArb
+
+		// If the channel has had its commitment broadcasted already,
+		// republish it in case it didn't propagate.
+		if !channel.HasChanStatus(channeldb.ChanStatusCommitBroadcasted) {
+			continue
+		}
+
+		closeTx, err := channel.BroadcastedCommitment()
+		switch {
+
+		// This can happen for channels that had their closing tx
+		// published before we started storing it to disk.
+		case err == channeldb.ErrNoCloseTx:
+			log.Warn("Channel %v is in state CommitBroadcasted, "+
+				"but no closing tx to re-publish...", chanPoint)
+			continue
+
+		case err != nil:
+			return err
+		}
+
+		log.Infof("Re-publishing closing tx for channel %v", chanPoint)
+		err = c.cfg.PublishTx(closeTx)
```

Comment:
Do we also need to manually set the channel arb's state to be `StateCommitmentBroadcasted `?

Reply:
This shouldn't be necessary, since the `chanArb` is already handling any closing tx appearing in chain regardless of the current state.

Worst case we save the tx before committing to `StateCommitmentBroadcasted`, and re-publish on startup. When the tx eventually confirms, we will treat it the same way as a "surprising" remote force close.